gnd/src/main/AndroidManifest.xml

```diff
@@ -70,18 +70,11 @@
       android:name="org.apache.http.legacy"
       android:required="false" />
 
-    <!--
-      Disable WorkManagerInitializer so that we can initialize ourselves in GndApplication setting
-      custom GndWorkerFactory to perform Dagger injection.
-      TODO(github.com/google/dagger/issues/1183): Remove this if and when Worker inject is supported
-      natively.
-    -->
+    <!-- Remove the default WorkManager initializer since we are providing our own -->
     <provider
       android:name="androidx.work.impl.WorkManagerInitializer"
-      android:authorities="com.google.android.gnd.workmanager-init"
-      android:enabled="false"
-      android:exported="false"
-      tools:replace="android:authorities" />
+      android:authorities="${applicationId}.workmanager-init"
+      tools:node="remove" />
```

Comment:
I am curious why this is needed? I understand that this is useful when we have to do some custom initialization, but I am not sure that we do. Do we?

Reply:
It's because we're using the `HiltWorkerFactory` from Jetpack (https://developer.android.com/training/dependency-injection/hilt-jetpack#workmanager) so we have to remove the default initializer.

Reply:
Based on the "WorkManager is not initialized properly" crash which I'm also getting, this may not be implemented correctly. Currently investigating.